src/components/ChatPanel.tsx

```diff
@@ -92,12 +92,20 @@ export function ChatPanel(props: ChatPanelProps) {
     res.catch((e) => (handleError ? handleError(e) : defaultHandleApiError(e)));
   }, [chat, props, messages, defaultHandleApiError, canSendMessage]);
 
-  const messagesRef = useRef<HTMLDivElement>(null);
+  const messagesRef = useRef<Array<HTMLDivElement | null>>([]);
+  const messagesContainer = useRef<HTMLDivElement>(null);
 
   // Scroll to the bottom of the chat when the messages change
   useEffect(() => {
-    messagesRef.current?.scroll({
-      top: messagesRef.current?.scrollHeight,
+    if (messagesRef?.current.length < 2) {
+      return;
+    }
+    const scrollTop = messagesRef.current
```

Comment:
could you add a little comment here on summing up the scroll heights of all the message elements except the last one?

Reply:
done